source/OptimaJet.DWKit.StarterApplication/Repositories/UserRepository.cs

```diff
@@ -6,17 +6,45 @@
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Logging;
 using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Services;
 
 namespace OptimaJet.DWKit.StarterApplication.Repositories
 {
     public class UserRepository : DefaultEntityRepository<User>
     {
+        public IOrganizationContext OrganizationContext { get; }
+        public ICurrentUserContext CurrentUserContext { get; }
         public UserRepository(
             ILoggerFactory loggerFactory,
             IJsonApiContext jsonApiContext,
+            IOrganizationContext organizationContext,
+            ICurrentUserContext currentUserContext,
             IDbContextResolver contextResolver
             ) : base(loggerFactory, jsonApiContext, contextResolver)
         {
+            this.OrganizationContext = organizationContext;
+            this.CurrentUserContext = currentUserContext;
+        }
+
+        public override IQueryable<User> Get()
+        {
+            if (OrganizationContext.InvalidOrganization) return base.Get().Take(0);
```

Comment:
semantically, what is an InvalidOrganization? what does this property represent?

anywho, to return an empty collection, I think you can just do 
```cs
 Enumerable.Empty<type of p>().AsQueryable();
```

Reply:
We tried that but got an exception (non async IQueryable?  Or something like that).  We put it back in and it worked.  Will change.

Reply:
InvalidOrganization is that we parsed a OrgId (as an integer), but it wasn’t found.  So returning empty.  If you want this to be the same as having an empty value (returning users in the same orgs as current user), that would be fine with me.